server/enterprise/message_export/shared/shared.go

```diff
@@ -66,6 +93,141 @@ type JobData struct {
 	TransferringZipMs      []int64
 	TotalBatchMs           []int64
 	Finished               bool
+	IsDownloadable         bool // exported
+}
+
+func JobDataToStringMap(jd JobData) map[string]string {
+	ret := make(map[string]string)
+	ret[JobDataExportType] = jd.ExportType
+	ret[JobDataExportDir] = jd.ExportDir
+	ret[JobDataBatchStartTime] = strconv.FormatInt(jd.BatchStartTime, 10)
+	ret[JobDataBatchStartId] = jd.BatchStartId
+	ret[JobDataJobStartTime] = strconv.FormatInt(jd.JobStartTime, 10)
+	ret[JobDataJobEndTime] = strconv.FormatInt(jd.JobEndTime, 10)
+	ret[JobDataJobStartId] = jd.JobStartId
+	ret[JobDataBatchSize] = strconv.Itoa(jd.BatchSize)
+	ret[JobDataChannelBatchSize] = strconv.Itoa(jd.ChannelBatchSize)
+	ret[JobDataChannelHistoryBatchSize] = strconv.Itoa(jd.ChannelHistoryBatchSize)
+	ret[JobDataBatchNumber] = strconv.Itoa(jd.BatchNumber)
+	ret[JobDataTotalPostsExpected] = strconv.Itoa(jd.TotalPostsExpected)
+	ret[JobDataMessagesExported] = strconv.Itoa(jd.MessagesExported)
+	ret[JobDataWarningCount] = strconv.Itoa(jd.WarningCount)
+	ret[JobDataIsDownloadable] = strconv.FormatBool(jd.IsDownloadable)
+	return ret
+}
+
+func StringMapToJobDataWithZeroValues(sm map[string]string) (JobData, error) {
```

Comment:
Is this mostly needed for tests or is there somewhere else we are planning to use it? Not sure I could spot it on the other side.

Reply:
It's used in the `worker.go` `extractJobData` fn -- I wanted to replace that old ad-hoc code with this more systematic (and tested) code. It's the fn above (`JobDataToStringMap`) which is only used in tests for now. I want to keep it for now because I think I might use it in the upcoming test PRs. But if I don't, I'll remove it (better to have less code, even if it's tested).

Reply:
In retrospect, it was a mistake to have `map[string]string` as model.Job.Data instead of `map[string]any`, or even a sum-type of potential JobData structs for each job type (but I digress).